### DISCUSSION TITLE
PowerPaint

### DISCUSSION THREAD

**Discussion by Tillerz** [vladmandic/sdnext#2618] [General]
Hi,

https://github.com/open-mmlab/mmagic/tree/main/projects/powerpaint

anyone know if this is a standalone thing or is this something that we could have as part of Controlnet? It has been posted in the controlnet inpaint+llama thread where they announced the inpaint+llama feature some time ago, which worked once and then broke. I really want to have a nice outpaint to fix images where stuff is cut off at one or more sides. :)

**Comment by vladmandic** [comment]
this is a separate project. anything can be integrated, but this would be massive task.

**Comment by zhuang2002** [comment]
We have recently developed a new version of [PowerPaint](https://github.com/zhuang2002/PowerPaint), drawing inspiration from Brushnet. The [Online Demo](https://openxlab.org.cn/apps/detail/rangoliu/PowerPaint) has been updated accordingly. In the coming days, we intend to make the model weights available as open source.

Feel free to try it out and give it a star!

The PowerPaint model possesses the ability to carry out diverse inpainting tasks, such as object insertion, object removal, shape-guided object insertion, and outpainting. Notably, we have retained the cross-attention layer that BrushNet had removed, which is essential for task prompt input.

|  | Object insertion | Object Removal|Shape-guided Object Insertion|Outpainting|
|-----------------|-----------------|-----------------|-----------------|-----------------|
| Original Image| ![cropinput](https://github.com/Sanster/IOPaint/assets/108931120/bf91a1e8-8eaf-4be6-b47d-b8e43c9d182a)|![cropinput](https://github.com/Sanster/IOPaint/assets/108931120/c7e56119-aa57-4761-b6aa-56f8a0b72456)|![image](https://github.com/Sanster/IOPaint/assets/108931120/cbbfe84e-2bf1-425b-8349-f7874f2e978c)|![cropinput](https://github.com/Sanster/IOPaint/assets/108931120/134bb707-0fe5-4d22-a0ca-d440fa521365)|
| Output| ![image](https://github.com/Sanster/IOPaint/assets/108931120/ee777506-d336-4275-94f6-31abf9521866)| ![image](https://github.com/Sanster/IOPaint/assets/108931120/e9d8cf6c-13b8-443c-b327-6f27da54cda6)|![image](https://github.com/Sanster/IOPaint/assets/108931120/cc3008c9-37dd-4d98-ad43-58f67be872dc)|![image](https://github.com/Sanster/IOPaint/assets/108931120/18d8ca23-e6d7-4680-977f-e66341312476)|
